### PR TITLE
Add Cylinder JWT/Splinter 0.5 support to app auth handler

### PR DESCRIPTION
**Pull request by shannynalayna**
This PR makes various updates to support the authorization requirements in Splinter 0.5. This updates the Splinter dep, and various others in support of this update, to 0.5 within the Grid daemon. Authorization, in the form of Cylinder JWTs, has been added to all requests sent to the Splinter backend. This also updates the Signer used by the app auth handler to a Cylinder signer, allowing the JWT to be formed from this signer and added to rest requests. 

This functionality is behind the experimental `cylinder-jwt-support` feature. 

**To test:** 
There are a few workarounds needed when using Splinter 0.5 with Grid when setting up the Splinter circuit. Make sure to clear out leftover containers before running the Splinter examples docker-compose file: 

```$ docker-compose -f examples/splinter/docker-compose.yaml up```  

Once you are able to access the splinter containers:

```$ docker-compose -f examples/splinter/docker-compose.yaml exec splinterd-alpha bash``` 

First, in order to make requests to the splinter CLI, we need to set up authorization for this node. 
You will need to add the alpha node client's public key, generated by the registry, to the Splinter's allow_keys file:

```root@splinterd-alpha:/# echo $(cat /registry/alpha.pub) >> /etc/splinter/allow_keys```

Once this is done, the alpha container is able to make authorized requests to Splinter. 
Running the command:
```root@splinterd-alpha:/# splinter circuit proposals -k /registry/alpha.priv -U http://splinterd-alpha:8085```
should return the empty proposal rows as we have not proposed a circuit yet. 

Once the Splinter client is authorized, you may propose a circuit. If using the Grid walkthrough, as Splinter 0.5 does not send the circuit's service `admin_keys` argument as a JSON string, you may see an error like this: 
```
splinterd-beta                     | [2021-08-06 20:34:24.462] T["consensus-admin::beta-node-000"] ERROR [splinter::consensus::two_phase] failed to accept proposal 35323037653039663537306561303533633537386637626466636632616135376136396330656433326236323137386134373262646365653939393237666563 due to error: failed to commit proposal: Unable to start service gsBB on circuit aSMq7-kqdED: failed to initialize service: invalid arguments specified: failed to parse admin_keys list: invalid number at line 1 column 2
```

Do not fret, this bug is recorded. In order to workaround this error, the admin keys must be formatted as a JSON string when proposing a circuit:

```
root@splinterd-alpha:/# splinter circuit propose --key /registry/alpha.priv --url http://splinterd-alpha:8085 --node alpha-node-000::tcps://splinterd-alpha:8044 --node beta-node-000::tcps://splinterd-beta:8044 --service gsAA::alpha-node-000 --service gsBB::beta-node-000 --service-type *::scabbard --management grid --service-arg *::admin_keys="[\"{Grid Daemon public key}\"]" --service-peer-group gsAA,gsBB
```

Once this is submitted, you should be able to see this proposal by running: 

```root@splinterd-alpha:/# splinter circuit proposals -k /registry/alpha.priv -U http://splinterd-alpha:8085```

Now, to create a circuit from this proposal similar steps must be taken in the Splinterd-beta container (or the peer container you are attempting to create a circuit with)
```$ docker-compose -f examples/splinter/docker-compose.yaml exec splinterd-beta bash``` 

Now, repeat the first step when connecting to the alpha node, and copy over the Beta node's public key to the allow_keys file:
```root@splinterd-beta:/# echo $(cat /registry/beta.pub) >> /etc/splinter/allow_keys```

Now, running:
```root@splinterd-beta:/# splinter circuit proposals -k /registry/beta.priv -U http://splinterd-beta:8085```
you should be able to see the circuit proposed by the alpha node.

At this point, you may vote to accept the circuit:
```root@splinterd-beta:/# splinter circuit vote {circuit ID} --accept -k /registry/beta.priv -U http://splinterd-beta:8085```

Once this is done, you should be able to see the circuit from either alpha or beta node, running:
```splinter circuit list -k /registry/{Node}.priv -U http://{Node host}:8085``` after replacing the relevant information.

At this point, you may continue with the Grid walkthrough as documented.

### CODE REVIEW

**Review by dnewh — APPROVED** [review]
tested

File: examples/splinter/docker-compose.yaml [hunk 1]
```diff
@@ -116,9 +116,16 @@ services:
       "
 
   generate-registry:
-    image: splintercommunity/splinter-cli:0.4
+    image: splintercommunity/splinter-cli:0.5
```
Comment (dplumb94) [inline examples/splinter/docker-compose.yaml:119]: Is this considered a dev image? I dont think we should depend on any dev images in Grid (unless we have a special docker-compose file for that).

**Review by dplumb94 — CHANGES_REQUESTED** [review]
I think we should keep the default docker example for Splinter using the stable release. We could create another one for testing 0.5.

**Comment by shannynalayna** [comment]
Rather than updating the daemon to an experimental Splinter release, we're going to wait until we have a clear upgrade path for Grid 0.2. Therefore, more tests must be conducted using Grid 0.2 as-is with Splinter 0.5/0.6 before we work on updating Grid's dependencies.

**Pull request closed without merging**